source/server/session_resource_repository.h

```diff
@@ -0,0 +1,221 @@
+#pragma once
```

Comment:
TODO remove pragma once :(. Google's style guide calls for include guards so I'll try to stick with that unless/until I can update everything.